File: java/tree/110_BalancedBinaryTree.java
class Solution {
    public int getHeight(TreeNode root){
        if (root == null) return 0;
        int L = getHeight(root.left);
        int R = getHeight(root.right);
        if ( (L==-1)||(R==-1)||(Math.abs(L-R)>1) )
            return -1;
        return Math.max(L,R)+1;
    }
    public boolean isBalanced(TreeNode root) {
        return getHeight(root)!=-1;
    }
}

File: python/tree/110_BalancedBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def Depth(self, root):
        if root is None: return 0; 
        L = self.Depth(root.left);
        R = self.Depth(root.right);
        if L==-1 or R==-1 or abs(L-R)>1: return -1;
        return 1+max(L,R);
    
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """    
        return self.Depth(root)!=-1;